packages/next/src/build/webpack/plugins/define-env-plugin.ts

```diff
@@ -177,7 +177,11 @@ export function getDefineEnv({
     'process.env.__NEXT_AFTER': config.experimental.after ?? false,
     'process.env.NEXT_DEPLOYMENT_ID': config.deploymentId || false,
     'process.env.__NEXT_FETCH_CACHE_KEY_PREFIX': fetchCacheKeyPrefix ?? '',
-    'process.env.__NEXT_MIDDLEWARE_MATCHERS': middlewareMatchers ?? [],
+    ...(isTurbopack
+      ? {}
+      : {
+          'process.env.__NEXT_MIDDLEWARE_MATCHERS': middlewareMatchers ?? [],
+        }),
```

Comment:
Isn't it possible to inject that as defined var into Turbopack too? Weird that we need two different code paths for Turbopack and webpack in page-loader.ts?

Reply:
The matchers are coming from middleware.ts, we do early static analysis for the webpack case but it's not ideal as it means these values are not tracked / cached across builds, want to make sure that we use the values that are already coming out of Turbopack. This is a temporary solution though, thinking of moving the matcher into buildManifest but that's not straightforward to do right now for the webpack case, so this is the in-between case that ensures it's functionally correct.